components/org.wso2.carbon.um.ws.service/src/main/java/org/wso2/carbon/um/ws/service/ClaimManagerService.java

```diff
@@ -99,9 +99,12 @@ public String getAttributeNameFromDomain(String domainName, String claimURI) thr
 
     public Claim getClaim(String claimURI) throws UserStoreException {
         try {
-            return  getClaimManager().getClaim(claimURI);
+            return  (Claim) getClaimManager().getClaim(claimURI);
```

Comment:
How about checking 'instance of' here before doing the cast and throwing a UserStoreException here?
Basically the same thing :smile:

Reply:
Here the type of the returning instance by getClaimManager().getClaim(claimURI) is really informative for debugging purposes in this case.
True that 'instance of' is light weight compared to throw exception. If we are to go with 'instance of' approach we should log the class type of the returned value. 
But IMO other details given in the stack trace will be important as well.
Since this is not that much-used service (This has been not working from 5.0.0 release), perf thing will not be that much of concern.
WDYT?

Reply:
The problem here was, in AXIS2Services, return type should be as same as the actual instance type.
If a generic type is used it gives an AXIS2Fault. Please see, "[WSDL] [SOAP-Services] org.apache.axis2.AxisFault: Mapping qname not found" in Developer's List.

Considering the context I would like to keep the ClassCastException instead of going with 'Instance of' approach.